### PR TITLE
add make target for observability operator

### PR DESCRIPTION
**Pull request by StevenTobin**
## What
https://issues.redhat.com/browse/MGDSTRM-1433

## Why
The Makefile currently supports installing stand alone grafana and prometheus which does not reflect the actual stack in use.

## How
Adds a Makefile recipe for installing the observability operator with instructions for use.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,14 +27,36 @@ The options are available as separate `make` targets from the `install` folder.
...
+|DEX_USERNAME | Username for accessing dex |
+
+If you want to point to a local or non-default observatorium instance, edit the URL in your fork of the [resource file](https://github.com/bf2fc6cc711aee1a0c2a/observability-resources-mk/blob/main/development/index.json) and use the OBSERVABILITY_RESOURCES_GH variable to instatiate the operator with that config.
+
+Once the make recipe has completed the observability operator catalogue and config is deployed in the *kafka-observability* namespace. 
+
+* Click on `Operatorhub` in this namespace under the `Operators` tab in the left nav bar.
+* Click the entry under `Provider Type` for `observability-operator-manifests`
+* Click on `observability-operator`
+* Click `install`
```
Comment (jeremyary) [inline README.md:49]: IMO, we should go ahead and create the `Namespace` (or Project, whichever) and the `Subscription` to automate this step as well.

File: install/resources/observability-operator/catalogue-source.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: observability-operator-manifests
+  namespace: kafka-observability
```
Comment (jeremyary) [inline install/resources/observability-operator/catalogue-source.yaml:5]: AFAICT, we should be creating this namespace/project programatically as well or all the following resource creations will fail.

**Review by jeremyary — CHANGES_REQUESTED** [review]

File: install/Makefile [hunk 3]
```diff
@@ -123,7 +134,29 @@ test/chaos:
...
+	@ sed -i 's,<OBSERVABILITY_RESOURCES_GH>,$(OBSERVABILITY_RESOURCES_GH),' ./$(BUILD_DIR)/observability-operator/configmap.yaml
+
+	@ -oc new-project $(OBSERVABILITY_NAMESPACE)
+	@oc apply -f ./resources/observability-operator/catalogue-source.yaml
+	@oc apply -f ./${BUILD_DIR}/observability-operator/configmap.yaml
+	@oc apply -f ./${BUILD_DIR}/observability-operator/secrets.yaml
+
+.PHONY: uninstall/observability
+uninstall/observability:
+	@oc delete project $(OBSERVABILITY_NAMESPACE)
```
[the remark this answers is not in the document]
Comment (StevenTobin) [reply install/Makefile:174]: I've fixed that now, it deletes the cr and then the namespace.

File: README.md [hunk 4]
```diff
@@ -27,14 +27,36 @@ The options are available as separate `make` targets from the `install` folder.
...
 ```
 
+### Installing observability operator
+
+The Observability Operator install requires some variables to be set
+
+|Variable | Use|
+|------------|-------|
+|OBSERVABILITY_RESOURCES_GH | Github URL for observability resources. Defaults to [observability-resources-mk](https://api.github.com/repos/bf2fc6cc711aee1a0c2a/observability-resources-mk/contents)|
+|ACCESS_TOKEN | Github access token to pull observability resources |
```
[the remark this answers is not in the document]
Comment (StevenTobin) [reply README.md:37]: Of course, I've added that info to the README now.

File: README.md (hunk 1 above)
Comment (StevenTobin) [reply README.md]: Yep, added that.

File: install/resources/observability-operator/catalogue-source.yaml (hunk 2 above)
Comment (StevenTobin) [reply install/resources/observability-operator/catalogue-source.yaml]: Added the operatorgroup and subscription so it now automagically provisions the operator.

**Review by jeremyary — APPROVED** [review]
LGTM, thanks for the changes!

**Comment by tonyxrmdavidson** [comment]
Ran the script from a mac. The sed command does not work when deploying locally from a mac. Does work when replaced with a gsed command.

**Pull request merged**